src/components/Dashboard.tsx

```diff
@@ -0,0 +1,158 @@
+      <div className="title">
+        <Header as="h5">{name}</Header>
+
+        {hasApplyPermission && (
+          <Button inverted size="small" as={Link} to={`/application/new?type=${code}`} primary>
+            {constants.BUTTON_DASHBOARD_NEW}
+          </Button>
+        )}
+      </div>
+
+      <div className="filters">
+        {filters.map((filter) => (
+          <FilterComponent key={filter.id} template={template} filter={filter} />
+        ))}
+      </div>
+
+      <ViewAll template={template} />
+    </div>
+  )
+}
+
+const FilterComponent: React.FC<{ template: TemplateDetails; filter: Filter }> = ({
+  template,
+  filter,
+}) => {
+  const templateType = template.code
+  const { loading, applicationCount } = useListApplications({
+    type: templateType,
+    perPage: 1,
+    ...filter.query,
+  })
+
+  const applicationListUserRole =
+    filter.userRole === PermissionPolicyType.Apply ? USER_ROLES.APPLICANT : USER_ROLES.REVIEWER
+
+  const constructLink = () =>
+    `/applications?type=${templateType}&user-role=${applicationListUserRole}&${Object.entries(
+      filter.query
+    )
+      .map(([key, value]) => `${key}=${value}`)
+      .join('&')}`
+
+  if (loading) return null
+  if (applicationCount === 0) return null
+
+  return (
+    <div className="filter">
+      <Link to={constructLink()}>
+        {filter.icon && (
+          <Icon style={{ color: filter.iconColor }} name={filter.icon as SemanticICONS} />
+        )}
+        {`${applicationCount} ${filter.title}`}
+      </Link>
+    </div>
+  )
+}
+
+const rolesDisplay: {
+  [key: string]: {
+    icon: SemanticICONS
+  }
+} = {
+  [PermissionPolicyType.Apply]: {
+    icon: 'edit',
+  },
+  [PermissionPolicyType.Review]: {
+    icon: 'gavel',
+  },
+  [PermissionPolicyType.Assign]: {
+    icon: 'user plus',
+  },
+}
+
+const ViewAll: React.FC<{ template: TemplateInList }> = ({ template: { code, permissions } }) => {
+  const applicantRoles = permissions.filter((type) => type === PermissionPolicyType.Apply)
+  const reviewerRoles = permissions.filter((type) => type !== PermissionPolicyType.Apply)
+
+  const renderLink = (permissionTypes: string[], applicationListRole: USER_ROLES) => {
+    if (permissionTypes.length === 0) return null
+    return (
+      <Link
+        key={applicationListRole}
+        className="view-all clickable"
+        to={`/applications?type=${code}&user-role=${applicationListRole}`}
+      >
+        <div style={{ display: 'flex', alignItems: 'center' }}>
+          {permissionTypes.map((policyType) => {
+            const roleDisplay = rolesDisplay[policyType]
+            return <Icon key={policyType} name={roleDisplay.icon} size="small" />
+          })}
+          {constants.BUTTON_DASHBOARD_VIEW_ALL}
+        </div>
+      </Link>
+    )
+  }
+
+  return (
+    <div className="view-all-area">
+      {renderLink(applicantRoles, USER_ROLES.APPLICANT)}
+      {renderLink(reviewerRoles, USER_ROLES.REVIEWER)}
```

Comment:
That's cool - and if we have different user-roles (for now it's really just the difference around these too) we can add more. I would like to have this user-role selection in a separated place, but I guess that what was already there was hard to use. It could be a nice refactoring issue to replace the existing generation of columns on the list to also use this simplified one: Applicant or Reviewer basically for all different combinations of review + assign permissions

Reply:
Yeah, it sounds like we only really need to two roles. Good point, the selection should be shared, I did look into trying to use existing code, but could't, without pretty big change